src/main/java/org/maxgamer/quickshop/util/Util.java

```diff
@@ -576,6 +576,37 @@ public static String getItemStackName(@NotNull ItemStack itemStack) {
         return result == null ? MsgUtil.getItemi18n(itemStack.getType().name()) : result;
     }
 
+    /**
+     * Get a String List of all enchants on an enchanted book
+     * 
+     * @param itemStack
+     * @return The names of enchants contained on the enchanted book
+     */
+    @NotNull
+    public static List<String> getBooksEnchantments(@NotNull ItemStack itemStack) {
+        List<String> enchants = new ArrayList<>();
+        if (itemStack.getType() == Material.ENCHANTED_BOOK) {
+            EnchantmentStorageMeta meta = (EnchantmentStorageMeta) itemStack.getItemMeta();
+            if (meta.hasStoredEnchants()) {
+                for (Enchantment enchant : meta.getStoredEnchants().keySet()) {
+                    enchants.add(enchant.getKey().getKey());
+                }
+            }
+        }
+        return enchants;
+    }
+
+    @NotNull
+    public static boolean listContainsString(@NotNull List<String> list, @NotNull String string) {
+        boolean stringFound = false;
+        if (!list.isEmpty()) {
+            for (String entry : list) {
+                if (entry.contains(string))
+                    stringFound = true;
+            }
+        }
+        return stringFound;
+    }
```

Comment:
It can be merged to single method: isBookEnchantmentsMatched, and care about case issue and i18n:
```
    public static boolean isBookEnchantmentsMatched(ItemStack itemStack,String nameToMatch){
        if(itemStack.getType()== Material.ENCHANTED_BOOK){
            ItemMeta itemMeta=itemStack.getItemMeta();
            if(itemMeta instanceof EnchantmentStorageMeta){
                EnchantmentStorageMeta enchantmentStorageMeta= ((EnchantmentStorageMeta) itemMeta);
                for (Enchantment enchantment : enchantmentStorageMeta.getEnchants().keySet()) {
                    nameToMatch = nameToMatch.toUpperCase(Locale.ROOT);
                    if(enchantment.getKey().getKey().toUpperCase(Locale.ROOT).contains(nameToMatch)
                            || MsgUtil.getEnchi18n(enchantment).toUpperCase(Locale.ROOT).contains(nameToMatch)){
                        return true;
                    }
                }
            }
        }
        return false;
    }
```

Reply:
All good, I was caring about case and i18n once before but at some point I accidently deleted and had to rewrite it all